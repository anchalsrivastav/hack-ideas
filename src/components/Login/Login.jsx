import React, { useState } from "react";
import { useHistory, useLocation } from "react-router";
import {Input} from "antd";
import {loginWrapper, inputAndButton, loginButton, errorMessageText} from "./Login.styles";

const Login = () => {
    const history = useHistory();
    const [userId, setUserId] = useState(null);
    const [errorMessage, setErrorMessage] = useState('')
    const location = useLocation();
    const hackList = location.state?.hackList ? location.state?.hackList : [];

    const handleLogin = () => {
        if (userId == null) {
            setErrorMessage('Please provide valid Employee Id to login')
        }
        else {
            localStorage.setItem("userId", userId?.toString());
            history.push({
                pathname: '/dashboard',
                state: {
                    hackList: hackList,
                    loggedInUserId: userId
                }
            })
        }
    }

    const handleChange = (event) => { setUserId(event.target.value) }
    return(
        <div style={loginWrapper}>
            <img alt="login" src="/login.png" height="100%" width="100%"/>
            <div style={inputAndButton}>
                <h1>Enter Employee Id</h1>
                <Input type="text" name="userid" aria-label="userid-input" onChange={handleChange}/>
                {errorMessage && <div style={errorMessageText}>{errorMessage}</div>}
                <div style={loginButton}>
                    <button onClick={handleLogin}> Login </button>
                </div>
            </div>
        </div>
    )
}

export default Login;
