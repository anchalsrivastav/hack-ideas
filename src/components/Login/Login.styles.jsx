export const loginWrapper = {
    display: 'grid',
    gridTemplateColumns: '1fr 1fr',
    alignItems: 'center'
}

export const inputAndButton = {
    marginLeft: '20%',
    marginRight: '20%'
}

export const loginButton = {
    marginTop: '10%'
}

export const errorMessageText = {
    color: 'red'
}