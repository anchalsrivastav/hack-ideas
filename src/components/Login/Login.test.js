import React from "react";
import Login from "./Login";
import { render, screen, fireEvent } from '@testing-library/react';

const mockHistoryPush = jest.fn();

jest.mock('react-router', () => ({
  ...jest.requireActual("react-router"),
  useHistory: () => ({
    push: mockHistoryPush
  }),
  useLocation: () => ({
      pathname: '/dashboard',
      state: {
        hackList: null,
        loggedInUserId: '11'
      }
  })
}));

describe('Login', () => {
  it('should renders login component', () => {
    const {container} = render(<Login />);
    const loginText = screen.getByText(/Enter Employee Id/i);
    const input = screen.getByLabelText("userid-input");
    expect(input).toBeInTheDocument();
    fireEvent.change(input, {target: {value: '11'}})
    expect(input.value).toBe('11');
    expect(loginText).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  })

  it('should redirect to dashboard on login button click', () => {
    const {getAllByText} = render(<Login />);
    const input = screen.getByLabelText("userid-input");
    fireEvent.change(input, {target: {value: '11'}})
    fireEvent.click(getAllByText(`Login`)[0]);
    expect(mockHistoryPush).toHaveBeenCalledWith({
      pathname: '/dashboard',
      state: {
        hackList: [],
        loggedInUserId: '11'
      }
    })
  })
})

