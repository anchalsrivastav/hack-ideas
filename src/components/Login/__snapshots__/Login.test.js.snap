// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should renders login component 1`] = `
<div>
  <div
    style="display: grid; grid-template-columns: 1fr 1fr; align-items: center;"
  >
    <img
      alt="login"
      height="100%"
      src="/login.png"
      width="100%"
    />
    <div
      style="margin-left: 20%; margin-right: 20%;"
    >
      <h1>
        Enter Employee Id
      </h1>
      <input
        aria-label="userid-input"
        class="ant-input"
        name="userid"
        type="text"
        value="11"
      />
      
      <div
        style="margin-top: 10%;"
      >
        <button>
           Login 
        </button>
      </div>
    </div>
  </div>
</div>
`;
